src/Explorer/Tabs/GalleryTab.tsx

```diff
@@ -21,24 +22,42 @@ interface GalleryTabOptions extends ViewModels.TabOptions {
  */
 export default class GalleryTab extends TabsBase {
   private container: Explorer;
+  private galleryAndNotebookViewerComponentProps: GalleryAndNotebookViewerComponentProps;
   public galleryAndNotebookViewerComponentAdapter: GalleryAndNotebookViewerComponentAdapter;
 
   constructor(options: GalleryTabOptions) {
     super(options);
-
     this.container = options.container;
-    const props: GalleryAndNotebookViewerComponentProps = {
+
+    this.galleryAndNotebookViewerComponentProps = {
       container: options.container,
       junoClient: options.junoClient,
       notebookUrl: options.notebookUrl,
       galleryItem: options.galleryItem,
       isFavorite: options.isFavorite,
-      selectedTab: GalleryViewerTab.OfficialSamples,
+      selectedTab: options.selectedTab,
       sortBy: SortBy.MostViewed,
       searchText: undefined,
     };
+    this.galleryAndNotebookViewerComponentAdapter = new GalleryAndNotebookViewerComponentAdapter(
+      this.galleryAndNotebookViewerComponentProps
+    );
+  }
+
+  public reset(options: GalleryTabOptions) {
+    this.container = options.container;
+
+    this.galleryAndNotebookViewerComponentProps.container = options.container;
+    this.galleryAndNotebookViewerComponentProps.junoClient = options.junoClient;
+    this.galleryAndNotebookViewerComponentProps.notebookUrl = options.notebookUrl;
+    this.galleryAndNotebookViewerComponentProps.galleryItem = options.galleryItem;
+    this.galleryAndNotebookViewerComponentProps.isFavorite = options.isFavorite;
+    this.galleryAndNotebookViewerComponentProps.selectedTab = options.selectedTab;
+    this.galleryAndNotebookViewerComponentProps.sortBy = SortBy.MostViewed;
+    this.galleryAndNotebookViewerComponentProps.searchText = undefined;
 
-    this.galleryAndNotebookViewerComponentAdapter = new GalleryAndNotebookViewerComponentAdapter(props);
+    this.galleryAndNotebookViewerComponentAdapter.reset();
```

Comment:
>reset [](start = 50, length = 5)

when does this get called?

Reply:
Every time a gallery tab is already open and we try to launch gallery tab. This could happen if someone clicks on Gallery item from left menu or when publishing a notebook to gallery completes.